signing-with-metamask/index.md

```diff
@@ -0,0 +1,89 @@
+
+## Methods of Signing Data With MetaMask
+
+We'll explore various ways of signing data using the provider injected by MetaMask as well as implementations in web3.js and ethers.js:
+
+eth_sign: This method is used to sign data with the Ethereum address's private key and returns the signature in the format of `"0x<r><s><v>"`.
+
+The `eth_sign` method is used to sign data and prove ownership of an Ethereum address. A limitation of this method is that it does not include any prefix before signing the data, which could lead to the signature not being recognized as valid by some smart contracts. It also poses a security risk, so it should be avoided.
+
+personal_sign: This method is also used to sign data and prove ownership of an Ethereum address, similar to `eth_sign,` but adds an "Ethereum Signed Message" prefix to the data before signing it. It is part of the JSON-RPC personal module, which is for personal account management. It signs data with a user's private key and returns the signature. 
+
+"`tsx
+window.ethereum.request({ method: 'personal_sign',  params: [msg, from],  });
+```
+
+Where `data` is the data to be signed, `address` is the address of the user signing the data and `callback` is an optional callback function to handle the result. Advantages of this method include the added security provided by the prefix, which allows the signature to be recognized as valid by more smart contracts. However, a limitation of this method is that it only supports UTF-8 encoded data. Developers should choose this method if they need to sign data that will be recognized as valid by more smart contracts, and the data is UTF-8 encoded.
+
+
+
+eth_signTypedDataV4: This method is used to sign typed data (a specific format for structured data specified by the EIP-712 standard) and prove ownership of an Ethereum address, and returns the signature in the format of `0x<r><s><v>`.
+
+"`tsx
+window.ethereum.request({ method: 'eth_signTypedData_v4', params: [address, data] })
+```
+
+Where `data` is the typed data object to be signed, `address` is that of the user signing the data, and `callback` is an optional callback function to handle the result. Advantages of this method include its support for typed data, which allows for more structured and organized data, and its support for the EIP-712 standard, which is widely recognized and supported by the Ethereum community. However, a limitation of this method is that it only supports typed data, which means it can't be used to sign just any type of data. Developers should choose this method if they need to sign typed data according to the [EIP-712](https://eips.ethereum.org/EIPS/eip-712) standard.
```

Comment:
>a limitation of this method is that it only supports typed data, which means it can't be used to sign just any type of data

This sentence is a bit confusing. A reader might get the impression that if their data isn't already "typed", then it can't be used here. But in the typical case, the user of this method would be the ones writing/generating these types.

Maybe it would be better to say something along the lines of.... this might be unsuitable or less useful for data that can't be described easily with types? Something along those lines.